apps/activemq.go

```diff
@@ -17,14 +17,15 @@ package apps
 import (
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+	"github.com/GoogleCloudPlatform/ops-agent/internal/secret"
 )
 
 type MetricsReceiverActivemq struct {
 	confgenerator.ConfigComponent `yaml:",inline"`
 
-	Endpoint                               string `yaml:"endpoint" validate:"omitempty,hostname_port|startswith=service:jmx:"`
-	Username                               string `yaml:"username" validate:"required_with=Password"`
-	Password                               string `yaml:"password" validate:"required_with=Username"`
+	Endpoint                               string        `yaml:"endpoint" validate:"omitempty,hostname_port|startswith=service:jmx:"`
+	Username                               string        `yaml:"username" validate:"required_with=Password"`
+	Password                               secret.String `yaml:"password" validate:"required_with=Username"`
```

Comment:
Are these fields ignored?

Reply:
I think so, I put up https://github.com/GoogleCloudPlatform/ops-agent/pull/1322 to get rid of them. Thought it was best to split that out to its own thing since its unrelated to the redaction. I think this is a bug - even when the username and password are currently specified, these ignored fields are set and the jmx struct ones are not. And as a result, the generated otel config doesn't actually use them.